core/ecschema-locaters/test/GetSchemas.test.ts

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import * as fs from "fs";
+import  * as path from "path";
+import { ECVersion, ISchemaLocater, Schema, SchemaContext, SchemaKey, SchemaMatchType } from "@bentley/ecschema-metadata";
+import { SchemaJsonFileLocater } from "../src/SchemaJsonFileLocater";
+
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+describe("Concurrent schema accesses", () => {
+  const asyncContext = new SchemaContext();
+  const syncContext = new SchemaContext();
+  const schemaKeys: SchemaKey[] = [];
+
+  before(() => {
+    const schemaFolder = path.join(__dirname, "assets", "JSON");
+    const locater = new SchemaJsonFileLocater();
+    locater.addSchemaSearchPath(schemaFolder);
+    asyncContext.addLocater(locater as unknown as ISchemaLocater);
```

Comment:
I think we can use locater directly here since SchemaJsonFileLocater implements ISchemaLocater as well.

Reply:
Removed unnecessary type assertions.